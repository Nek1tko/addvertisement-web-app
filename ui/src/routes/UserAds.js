import React, { useEffect, useState } from "react";
import AdRecordsTable from "../components/AdRecordsTable";
import AuthService from '../services/auth.service';
import { Redirect } from "react-router-dom";
import axios from 'axios';
import authHeader from "../services/auth-header";

const API_URL = "http://localhost:8080/ad/user/";

const AdRecordsTableWrapper = props => {
    const [ads, setAds] = useState([]);
    const { _, userId, } = AuthService.getUser();

    useEffect(() => {
        axios
            .get(API_URL + userId, { headers: authHeader() })
            .then(res => {
                console.log(res);
                setAds(res.data);
            })
    }, [userId, setAds]);

    return <AdRecordsTable
        ads={ads}
        userId={userId}
        history={props.history}
    />;
}

const UserAds = props => {
    if (!AuthService.getUser()) {
        return <Redirect to="/login" />
    }

    return <AdRecordsTableWrapper history={props.history} />
};

export default UserAds;