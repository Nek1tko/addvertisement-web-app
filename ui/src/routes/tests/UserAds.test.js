/**
 * @group unit
 */

import { createBrowserHistory } from "history";
import AuthService from "../../services/auth.service";
import authHeader from "../../services/auth-header";
import axios from "axios";
import React from "react";
import { mount } from "enzyme";
import { Router } from "react-router-dom";
import UserAds from "../UserAds";
import { act } from "react-dom/test-utils";

jest.mock('axios');
jest.mock("../../services/auth.service");
jest.mock("../../services/auth-header");

const mCategory = {
    id: 1,
    name: "mock category"
};

const mSubcategory = {
    id: 1,
    name: "mock subcategory",
    category: mCategory
};

const mMetro = {
    id: 1,
    name: "Улица Дыбенко",
};

const mAd = {
    id: 1,
    description: "mock description",
    price: "123",
    name: "mock name",
    metro: mMetro,
    subcategory: mSubcategory,
    is_active: true,
    isFavourite: true,
};

describe('UserAdsTests', () => {
    let wrapper, props;
    const setState = jest.fn();

    beforeEach(() => {
        props = {
            history: {
                ...createBrowserHistory(),
                push: jest.fn()
            }
        };
        AuthService.getUser.mockReturnValue({ userId: "mockUser" });
        authHeader.mockReturnValue({ header: "mockHeader" });
        axios.get.mockResolvedValue({
            data: [
                mAd,
                mAd
            ]
        });
        const useStateSpy = jest.spyOn(React, 'useState');
        useStateSpy.mockImplementation((state) => [state, setState]);
        wrapper = mount(
            <Router history={props.history}>
                <UserAds {...props} />
            </Router>);
    });

    afterEach(() => {
        jest.clearAllMocks();
    });

    test('UserAdsRenderTest', () => {
        act(() => {
            mount(<Router history={props.history}>
                <UserAds {...props} />
            </Router>);
        });
    });

    test('UserAdsSnapshotTest', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('UserAdsNotAuthorizedTest', () => {
        AuthService.getUser.mockReturnValue(null);
        wrapper = mount(
            <Router history={props.history}>
                <UserAds {...props} />
            </Router>);
    });
});
