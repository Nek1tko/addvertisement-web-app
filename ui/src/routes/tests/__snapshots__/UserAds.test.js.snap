// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserAdsTests UserAdsSnapshotTest 1`] = `ReactWrapper {}`;
